app/src/main/java/com/iyehuda/feelslike/ui/newpost/NewPostFragment.kt

```diff
@@ -0,0 +1,73 @@
+package com.iyehuda.feelslike.ui.newpost
+
+import android.net.Uri
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import androidx.navigation.fragment.findNavController
+import com.iyehuda.feelslike.databinding.FragmentNewPostBinding
+
+class NewPostFragment : Fragment() {
+
+    private var _binding: FragmentNewPostBinding? = null
+    private val binding get() = _binding!!
+
+    private lateinit var viewModel: NewPostViewModel
+
+    // Register the image picker ActivityResultLauncher
+    private val imagePickerLauncher = registerForActivityResult(ActivityResultContracts.GetContent()) { uri: Uri? ->
```

Comment:
Use the `ImagePicker` utility here (see usage in `SignupFragment` and `EditProfileFragment`)